server/lyft/feature/github_client.go

```diff
@@ -0,0 +1,105 @@
+package feature
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+type githubClient interface {
+	GetContents(owner, repo, branch, path string) ([]byte, error)
+}
+
+type RepoConfig struct {
+	Owner  string
+	Repo   string
+	Branch string
+	Path   string
+}
+
+// CustomGithubRetriever uses Atlantis' internal client to retrieve the contents
+// of the feature file.  This allows us to re-use GH credentials easily as opposed
+// to the default ffclient.GithubRetriever.
+type CustomGithubRetriever struct {
+	Client githubClient
+	Cfg    RepoConfig
+}
+
+func (c *CustomGithubRetriever) Retrieve(ctx context.Context) ([]byte, error) {
+	return c.Client.GetContents(c.Cfg.Owner, c.Cfg.Repo, c.Cfg.Branch, c.Cfg.Path)
+}
+
+type installationFetcher interface {
+	FetchInstallationToken(ctx context.Context) (int64, error)
+}
+
+type InstallationFetcher struct {
+	ClientCreator githubapp.ClientCreator
+	Org           string
+}
+
+func (i *InstallationFetcher) FetchInstallationToken(ctx context.Context) (int64, error) {
+	appClient, err := i.ClientCreator.NewAppClient()
+	if err != nil {
+		return 0, errors.Wrap(err, "creating app client")
+	}
+
+	installation, _, err := appClient.Apps.FindOrganizationInstallation(ctx, i.Org)
+	if err != nil {
+		return 0, errors.Wrapf(err, "finding organization installation")
+	}
+	return installation.GetID(), nil
```

Comment:
why don't u reuse this code that already exists here? https://github.com/lyft/atlantis/blob/6d8d52eea9aea773737e98441e26a072b3805c7e/server/vcs/provider/github/installation.go#L18-L19

Reply:
See comment on import cycle.